File: f1app/src/main/java/com/f1/f1app/service/ForumService.java
package com.f1.f1app.service;

import com.f1.f1app.dao.impl.ForumDaoImpl;
import com.f1.f1app.model.Discussion;
import com.f1.f1app.model.Topic;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ForumService {

    @Autowired
    private ForumDaoImpl forumDao;

    public List<Topic> getAll(){
        return this.forumDao.getAll();
    }

    public Topic addNew(String topic){
        return this.forumDao.addNew(topic);
    }

    public List<Discussion> getDiscussionForTopic(Long topicId){
        return this.forumDao.getDiscussionForTopic(topicId);
    }

    public Discussion updateDiscussion(Long discussionId, String newComment){
        return this.forumDao.updateDiscussion(discussionId, newComment);
    }

    public boolean deleteComment(Long id){
        return this.forumDao.deleteComment(id);
    }

    public Discussion addNewDiscussion(Discussion d){
        return this.forumDao.addNewDiscussion(d);
    }
}

File: client/f1-app/src/app/core/services/forum.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Topic } from "../model/topic.model";
import { Discussion } from "../model/discussion.model";

@Injectable({
  providedIn: 'root'
})
export class ForumService {


  constructor(protected http: HttpClient){}

  public getAllTopics() {
    return this.http.get<{ success: boolean, topics: Topic[] }>("http://localhost:8080/forum/getAll");
  }

  public addNewTopic(topic: string) {
    return this.http.post<{ success: boolean, topic: Topic }>(`http://localhost:8080/forum/newTopic?newTopic=${topic}`, {}
     );
  }

  public getDiscussionForTopic(topicId: number) {
    return this.http.get<{ success: boolean, discussions: Discussion[] }>(`http://localhost:8080/forum/getDiscussionForTopic/${topicId}`);
  }

  public updateComment(comment: Discussion) {
    return this.http.post<{ success: boolean, discussion: Discussion }>(`http://localhost:8080/forum/updateDiscussion`, {discussionId: comment.id, newComment: comment.comment});
  }

  public deleteComment(commentId: number) {
    return this.http.delete<{ success: boolean }>(`http://localhost:8080/forum/deleteComment/${commentId}`);
  }

  public addNewDiscussion(discussion: Discussion) {
    return this.http.post<{ success: boolean, newDiscussion: Discussion }>(`http://localhost:8080/forum/addNew`, {...discussion});
  }



}